src/zoomifytilesource.js

```diff
@@ -0,0 +1,97 @@
+(function( $ ){
+
+  /**
+   * @class ZoomifyTileSource
+   * @classdesc A tilesource implementation for the zoomify format.
+   *
+   * @memberof OpenSeadragon
+   * @extends OpenSeadragon.TileSource
+   * @param {Number} width - the pixel width of the image.
+   * @param {Number} height
+   * @param {Number} tileSize
+   * @param {String} tilesUrl
+   */
+  $.ZoomifyTileSource = function( options ) {
+
+  	options.tileSize = 256;
+		
+		var currentImageSize = {x:options.width,y:options.height};
+		options.imageSizes = [{x:options.width,y:options.height}];
+  	options.gridSize = [this._getGridSize(options.width,options.height,options.tileSize)];
+		
+  	while(parseInt(currentImageSize.x) > options.tileSize || parseInt(currentImageSize.y) > options.tileSize) {
+				currentImageSize.x = Math.floor(currentImageSize.x/2);
+				currentImageSize.y = Math.floor(currentImageSize.y/2);
+				options.imageSizes.push({x:currentImageSize.x,y:currentImageSize.y});
+				options.gridSize.push(this._getGridSize(currentImageSize.x,currentImageSize.y,options.tileSize));
+			}
+		options.imageSizes.reverse();
+		options.gridSize.reverse();
+		options.minLevel = 0;
+		options.maxLevel = options.gridSize.length-1;
+		
+  	
+  	OpenSeadragon.TileSource.apply( this, [ options ] );
+
+  };
+
+  $.extend( $.ZoomifyTileSource.prototype, $.TileSource.prototype, /** @lends OpenSeadragon.ZoomifyTileSource.prototype */{
+		
+		_getGridSize: function(width,height,tileSize) {
```

Comment:
Yes, we prefer this form of identifying private functions (prefixing with underscore). Also please add `// private` above the function.